### ISSUE TITLE
Wrong error message on validation of stim file

### ISSUE THREAD

**Issue by Remi-Gau** [bids-standard/legacy-validator#1925]
This relates to this dataset: https://openneuro.org/datasets/ds000238/versions/00002

This dataset contains "stim" files:

```
/sub-01/beh/sub-01_task-test_stim.tsv.gz
```

That have several problems:

- they should actually be `beh.tsv` files as they are recording suibject's response and not stimuli related data
- even if they were stim files
  - they contain a header where as stim.tsv.gz should not
  - they do not have the required stim.json side car

Also I think that the error reported by the openneuro validator is not quite what I would expect. 

Error pasted below:

```
Location:

/sub-01/beh/sub-01_task-test_stim.tsv.gz

Reason:

You should define 'ArterialSpinLabelingType' for this file. 'ArterialSpinLabelingType' can be CASL, PCASL, PASL.
```

**Comment by effigies** [comment]
It is a weird report. If you download it though, you get:

```
❯ bids-validator --verbose --remoteFiles .
bids-validator@1.14.5
(node:691621) Warning: Closing directory handle on garbage collection
(Use `node --trace-warnings ...` to show where the warning was created)
	1: [ERR] Continous recording data files are required to have an associated JSON metadata file. (code: 170 - CONTINOUS_RECORDING_MISSING_JSON)
		./sub-01/beh/sub-01_task-test_stim.tsv.gz
			Continous recording data files are required to have an associated JSON metadata file.

	Please visit https://neurostars.org/search?q=CONTINOUS_RECORDING_MISSING_JSON for existing conversations about this issue.

	1: [WARN] Tabular file contains custom columns not described in a data dictionary (code: 82 - CUSTOM_COLUMN_WITHOUT_DESCRIPTION)
		./participants.tsv
			Tabular file contains custom columns not described in a data dictionary
			Evidence: Columns: age, sex, handedness, design_order1, design_order2, design_order3, design_order4, design_order5 not defined, please define in: /participants.json
		./sub-01/func/sub-01_task-med8_run-1_events.tsv
			Tabular file contains custom columns not described in a data dictionary
			Evidence: Columns: stimulus, Var4 not defined, please define in: /events.json, /task-med8_events.json,/run-1_events.json,/task-med8_run-1_events.json,/sub-01/sub-01_events.json,/sub-01/sub-01_task-med8_events.json,/sub-01/sub-01_run-1_events.json,/sub-01/sub-01_task-med8_run-1_events.json,/sub-01/func/sub-01_events.json,/sub-01/func/sub-01_task-med8_events.json,/sub-01/func/sub-01_run-1_events.json,/sub-01/func/sub-01_task-med8_run-1_events.json

	Please visit https://neurostars.org/search?q=CUSTOM_COLUMN_WITHOUT_DESCRIPTION for existing conversations about this issue.

        Summary:                  Available Tasks:        Available Modalities: 
        14 Files, 187.04MB        Med8                    MRI                   
        1 - Subject               Slow12                                        
        1 - Session               Quick4n                                       
                                  Slow10                                        
                                  Quick6                                        


	If you have any questions, please post on https://neurostars.org/tags/bids.
```

So this is likely a problem where OpenNeuro stored a buggy validation report long-term.
